src/test/ui/suggestions/missing-trait-bounds-for-method-call.stderr

```diff
@@ -0,0 +1,44 @@
+error[E0599]: no method named `foo` found for reference `&Foo<T>` in the current scope
+  --> $DIR/missing-trait-bounds-for-method-call.rs:14:14
+   |
+LL | struct Foo<T> {
+   | ------------- doesn't satisfy `Foo<T>: Bar`
+...
+LL | trait Bar {
+   | --------- `Bar` defines an item `foo`, perhaps you need to implement it
+...
+LL |         self.foo();
+   |              ^^^ method not found in `&Foo<T>`
```

Comment:
I think this should go before both the struct and trait above

Reply:
@tmandry What do you think of this?

<img width="627" alt="Screen Shot 2020-02-24 at 6 53 40 PM" src="https://user-images.githubusercontent.com/1606434/75210811-03652700-5737-11ea-9ae3-e0e60e75121c.png">

Reply:
This looks really, really good!

My only critique is that the two `note`s (one on the span, one outside) are redundant.

Reply:
In the new commit, this test looks like this now:

<img width="619" alt="Screen Shot 2020-02-24 at 10 38 59 PM" src="https://user-images.githubusercontent.com/1606434/75224741-e7c14700-575d-11ea-861e-2d5cc613f163.png">

It is a very niche case, but with this we handle it well enough, I think.

Reply:
Great! And I forgot to say again that I think the order of the spans ought to be reversed :)